tsconfig-for-webpack.json

```diff
@@ -1,7 +1,16 @@
 {
+  "extends": "./tsconfig.base.json",
+  "ts-node": {
+    // This can be omitted when using ts-patch
+    "compiler": "ttypescript",
+    "moduleTypes": {
+      "webpack/*": "cjs",
+      // Globs are also supported with the same behavior as tsconfig "include"
+      "webpack/**/*": "cjs"
+    }
+  },
   "compilerOptions": {
-    "module": "commonjs",
-    "target": "es5",
-    "esModuleInterop": true
+    "moduleResolution": "node",
+    "skipLibCheck": false
```

Comment:
Can you explain why the compiler options are changing here and why we need extra settings for `webpack/*`?

I also think that we should `skipLibCheck: true` here until we also make it true in `tsconfig.base.json` so that we do the lib check everywhere.  I would prefer not to only fail in webpack and also fail when building.  I would suggest to skip the lib check now, and we can have another PR that turns it on everywhere.

Reply:
> 

in node14, it works fine if a cjs try to require a esm(.mjs or .ts file), but it doesn't work in node 16(or I say ts-node v0.10)

`target` and  `module` set to `"es2020“` or `”esnext“` is recommand by ts-node for configuraing webpack.

Reply:
Thanks for all the explanation, I really appreciate it.  That all sounds good to me.  :+1: